cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -109,17 +128,56 @@ func getCertForOauthID(priv *ecdsa.PrivateKey, scp signingCertProvider, connecto
 
 	resp, err := scp.SigningCert(params, bearerAuth)
 	if err != nil {
-		return nil, nil, err
+		return Resp{}, err
+	}
+	sct, err := base64.StdEncoding.DecodeString(resp.SCT.String())
+	if err != nil {
+		return Resp{}, err
 	}
 
 	// split the cert and the chain
 	certBlock, chainPem := pem.Decode([]byte(resp.Payload))
-	certPem = pem.EncodeToMemory(certBlock)
-	return certPem, chainPem, nil
+	certPem := pem.EncodeToMemory(certBlock)
+	fr := Resp{
+		CertPEM:  certPem,
+		ChainPEM: chainPem,
+		SCT:      sct,
+	}
+
+	// verify the sct
+	if err := VerifySCT(fr); err != nil {
+		return Resp{}, errors.Wrap(err, "verifying SCT")
+	}
+	fmt.Println("Successfully validated SCT...")
+	return fr, nil
+}
+
+// verifySCT verifies the SCT against the Fulcio CT log public key
+// The SCT is a `Signed Certificate Timestamp`, which promises that
+// the certificate issued by Fulcio was also added to the public CT log within
+// some defined time period
+func verifySCT(fr Resp) error {
+	buf := tuf.ByteDestination{Buffer: &bytes.Buffer{}}
+	if err := tuf.GetTarget(context.TODO(), ctPublicKeyStr, &buf); err != nil {
```

Comment:
not really sure if we should error out on this -- because it will fail if you haven't run cosign init and i don't think we want that to be failure behavior (rather preferrable behavior)

Reply:
sure, i can add in a warning that SCTs won't be verified if the user doesn't run `cosign init` 

in the future i wonder if we could just automatically run `cosign init` for someone if they have experimental mode set!

Reply:
Maybe a silly question, but If we have all the info to validate against in the binary to run "cosign init", could we just validate it there before putting it on disk?

Reply:
"there"? 
i think i'm having trouble parsing the question! do you mean if you haven't run cosign init at this point, go ahead and run it?